internal/kconfig/kconfig.go

```
@@ -263,12 +263,25 @@ func PutInteger(data []byte, integer *btf.Int, n uint64) error {
 		return fmt.Errorf("invalid boolean value: %d", n)
 	}
 
+	if len(data) < int(integer.Size) {
+		return fmt.Errorf("can't fit an integer of size %d into a byte slice of length %d", integer.Size, len(data))
+	}
+
 	switch integer.Size {
 	case 1:
+		if integer.Encoding == btf.Signed && (n > math.MaxInt8 && n < math.MaxUint64-math.MaxInt8) {
```

Sorry, but again I don't understand why it needs to be so complicated:
```suggestion
		if integer.Encoding == btf.Signed && (n > math.MaxInt8 || n < math.MinInt8) {
```

---

This is because n is of type uint and MinInt8 is of type int, so there was a compiler error in the comparison operation. However, when I cast n to int64, I saw that there was no problem when n exceeded MaxInt64 of int64. I will create a pull request with a similar format by converting n to int64.

---

> I will create a pull request with a similar format by converting n to int64.

Please, no more pull requests. Just amend and force-push to this PR's branch, please.

---

I forced push.